### PR TITLE
Add support for django 1.8

### PR DESCRIPTION
**Pull request by jlward**

### CODE REVIEW

File: attachments/tests.py [hunk 1]
```diff
@@ -1,92 +1,103 @@
...
+from django.core.urlresolvers import reverse
+from django.db import models
+from django.test import TestCase
 
-from attachments.models import Attachment, TestModel
-
-import os
-from tempfile import NamedTemporaryFile
-
-"""
```
Comment (jlward) [inline attachments/tests.py]: This doc test guy was removed.

File: attachments/tests.py [hunk 2]
```diff
@@ -1,92 +1,103 @@
...
-<Attachment: Something>
+from attachments.models import Attachment
 
 
->>> att2 = Attachment.objects.create_for_object(
-...     tm, file=TEST_FILE2, attached_by=bob, title="Something Else",
-...     summary="Something else more")
->>> att2
-<Attachment: Something Else>
+class TestModel(models.Model):
```
Comment (jlward) [inline attachments/tests.py]: Moved this guy out of `attachment.models.py` since it's only needed for tests.

File: attachments/urls.py [hunk 3]
```diff
@@ -1,9 +1,13 @@
-from django.conf.urls.defaults import *
+try:
+    from django.conf.urls import patterns, url
+except ImportError:  # Django 1.4
```
Comment (jlward) [inline attachments/urls.py]: This was needed

File: attachments/views.py [hunk 4]
```diff
@@ -1,21 +1,26 @@
+import json
```
Comment (jlward) [inline attachments/views.py]: This was needed.

File: attachments/views.py [hunk 5]
```diff
@@ -95,14 +105,5 @@ def list_attachments(request, content_type, object_id, order_by=None,
     if order_by:
         attachments = attachments.order_by(*order_by)
 
-    for media_type in request.accepted_types:
```
Comment (jlward) [inline attachments/views.py]: This never worked.

File: attachments/urls.py [hunk 6]
```diff
@@ -1,9 +1,13 @@
...
+    from django.conf.urls.defaults import patterns, url
 from django.contrib import admin
 
 admin.autodiscover()
 
-urlpatterns = patterns('attachment.views',
```
Comment (jlward) [inline attachments/urls.py]: Notice this was `attachment`. Which means this never worked either.

File: run_tests.py [hunk 7]
```diff
@@ -0,0 +1,37 @@
+#!/usr/bin/env python
```
Comment (jlward) [inline run_tests.py]: I'm really glad I found a `run_tests.py` script that I like and is easy to drop into other apps.

File: attachments/tests.py [hunk 8]
```diff
@@ -1,92 +1,103 @@
...
+
+    @contextmanager
+    def open(self, text=None):
+        if text is None:
+            text = 'some test text'
+        with NamedTemporaryFile('w+') as f:
+            f.write(text)
+            f.flush()
+            f.seek(0)
+            yield File(f)
```
Comment (bocribbz) [inline attachments/tests.py]: Personal curiosity, why `yield` here?

Comment (jlward) [inline attachments/tests.py]: Because it's a context manager, and this is the guy I want in the context.

Comment (bocribbz) [inline attachments/tests.py]: so `return File(f)` would close the file and the file would disappear?

Comment (jlward) [inline attachments/tests.py]: `return` does weird things in a context manager.

File: attachments/tests.py [hunk 9]
```diff
@@ -1,92 +1,103 @@
...
+    def test_view_smoke_test(self):
+        url = reverse(
+            'attachment_list',
+            kwargs={
+                'content_type': self.content_type.pk,
+                'object_id': self.tm.pk,
+            },
+        )
+        r = self.client.get(url)
+        self.assertEqual(r.status_code, 200)
```
Comment (bocribbz) [inline attachments/tests.py]: Dont we want to test a list page with some attachments too?

File: attachments/tests.py [hunk 10]
```diff
@@ -1,92 +1,103 @@
...
+            summary="Something",
+        )
+        url = reverse(
+            'attachment_delete',
+            kwargs={
+                'attachment_id': attachment.pk,
+            },
+        )
+        r = self.client.post(url)
+        self.assertEqual(r.status_code, 200)
```
Comment (bocribbz) [inline attachments/tests.py]: isnt it a bad idea to have a 200 response on a POST request that goes through?

File: attachments/tests.py (hunk 9 above)
Comment (jlward) [inline attachments/tests.py]: I was only trying to add tests would result in all the files being loaded (showing us import errors). I checked all the views to make sure everything was kosher in the them.

File: attachments/tests.py (hunk 10 above)
Comment (jlward) [inline attachments/tests.py]: The goal of this ticket is not to fix django-attachments, but to add support for django 1.8. If I spent the time to fix every lib I was upgrading, we would never get the upgrade to django 1.8 done.

File: attachments/views.py [hunk 11]
```diff
@@ -1,21 +1,26 @@
...
+    request,
+    content_type,
+    object_id,
+    template_name='attachments/new_attachment.html',
+    form_cls=AttachmentForm,
+    redirect=lambda object, attachment: object.get_absolute_url(),
+):
+    object_type = get_object_or_404(ContentType, id=int(content_type))
     try:
         object = object_type.get_object_for_this_type(pk=int(object_id))
```
Comment (bocribbz) [inline attachments/views.py]: uff `object`... yeah i know its not in this ticket's scope

Comment (jlward) [inline attachments/views.py]: Yea, it's not good. It's actually pretty concerning that attachments are such a huge part of PS and the code is so bad.

File: attachments/views.py [hunk 12]
```diff
@@ -78,13 +87,14 @@ def delete_attachment(request, attachment_id, redirect=None):
...
     else:
         message = {'success': True}
-        content = simplejson.dumps(message, ensure_ascii=False)
+        content = json.dumps(message, ensure_ascii=False)
         return HttpResponse(content, content_type='application/json')
 
+
 @login_required
 def list_attachments(request, content_type, object_id, order_by=None,
                      template_name='attachments/list_attachments.html'):
```
Comment (bocribbz) [inline attachments/views.py]: `template_name` no longer in use?

File: run_tests.py (hunk 7 above)
Comment (bocribbz) [inline run_tests.py]: :+

**Comment by bocribbz** [comment]
looks good :+1:

**Pull request merged**